### DISCUSSION TITLE
Script to launch project with multiple windows and shells

### DISCUSSION THREAD

**Discussion by vlopezferrando** [i3/i3#5859] [Show and tell]
Hello! As someone who usually works on many projects at the same time, I find it useful to quickly launch multiple programs on a workspace that leave me a perfectly working setup for that project.

In my case, this usually means a code editor, a browser, and a bunch of shells with some commands run on each shell.

Today I finished the last iteration of the script I use for this purpose, and I thought of sharing it here in case someone finds it useful. Any tip is also welcome.

You will see I decided to build the layout manually (splitting, resizing), instead of saving the layout tree. I found it easier to have it all as a single script.

This is the result:

![2024-01-08 16-56-19](https://github.com/i3/i3/assets/1635719/2367d4cc-862a-4004-884b-53d433bf2eae)

And this is the script:

```bash
#!/bin/bash

DIR=$HOME/git/python-cards

# Tip: use `xprop` to find out the class of any window.
function waitfor() {
  xdotool search --sync --all --onlyvisible --desktop $(xprop -notype -root _NET_CURRENT_DESKTOP | cut -c 24-) $@
}

function type_text() {
  # Tip: Copy and paste from clipboard instead of typing the text
  echo "$2" | xclip -selection clipboard
  xdotool search --name $1 key --window %@ ctrl+shift+V
  xdotool search --name $1 key --window %@ Return
}

# Open VS Code
code --new-window $DIR
waitfor --class code

# Open Firefox on the right
i3-msg split h
firefox -new-window localhost:8000 &
waitfor --class firefox

# Open three terminals below Firefox

# Shell
i3-msg split v
alacritty --title shell -e zsh -c "cd $DIR && zsh" &
waitfor --name shell
type_text shell ". ./venv/bin/activate"

# Runserver
i3-msg split v
alacritty --title "python-cards-runserver" -e zsh -c "cd $DIR && zsh" &
waitfor --name runserver
type_text runserver ". ./venv/bin/activate"
type_text runserver "./manage.py runserver"

# Tailwind
i3-msg split v
alacritty --title "python-cards-tailwind" -e zsh -c "cd $DIR && zsh" &
waitfor --name tailwind
type_text tailwind ". ./venv/bin/activate"
type_text tailwind "nvm use 14 && ./manage.py tailwind start"

# Shrink shells
i3-msg focus parent
i3-msg focus parent
i3-msg focus parent
i3-msg resize shrink up 200 px

# Reload Firefox
i3-msg focus up
sleep 1
xdotool key F5
```

**Comment by qrockz** [comment]
`xprop -notype -root _NET_CURRENT_DESKTOP | cut -c 24-` gives me 2 but I'm on workspace 8

![outtt](https://github.com/i3/i3/assets/56853194/7f44fcc1-5afa-4200-994d-3955dfd240ae)

I'm using 3 monitors.

**Reply by vlopezferrando** [reply]
I believe

```bash
xprop -notype -root _NET_CURRENT_DESKTOP | cut -c 24-
```

Gives you the index of the current active _monitor_.
